packages/native/src/useLinking.native.tsx

```diff
@@ -122,3 +122,18 @@ export default function useLinking(
     getInitialState,
   };
 }
+
+function checkIsPrefixMatch(
+  url: string,
+  prefix: string | RegExp
+): string | RegExp | false {
+  if (typeof prefix === 'string') {
+    if (url.startsWith(prefix)) {
+      return prefix;
+    }
+  } else if (prefix.test(url)) {
```

Comment:
I'd avoid `regex.test` since `test` is stateful when used with `/g` and this regex is provided from outside the lib.
```suggestion
  } else if (url.match(prefix)) {
```